docs/ga-roadmap.md

```diff
@@ -0,0 +1,114 @@
+
+## Current Status
+
+The OpenTelemetry Collector ecosystem right now has a lot of components that are in different stages of stability (experimental, alpha, beta, etc.). All these components have different public APIs/Interfaces (e.g. code API, configuration, etc.).
+
+A significant amount of legacy code was inherited from the Collector's ancestor [OpenCensus Service](https://github.com/census-instrumentation/opencensus-service), since then the Collector changed the internal data model and other significant changes were made.
+
+Trying to mark the entire ecosystem GA, at the same moment, will be a significant effort and will take a significant amount of time.
+
+## Proposal
+
+This document proposes a GA Roadmap based on multiple phases, where different parts of the collector will be released as stable at different moments of time.
+
+At this moment we are defining only the first two phases of the process, and the next phases will be defined at a later stage once the Collector maintainers will have better understanding of the process and implications.
+
+When other signals data models (proto definition) will be marked as stable,
+
+### Phase 1
+
+**Tentative Date:** 2021-03-31
+
+**KR:** At the end of this phase the Collector’s core API/SPI will be marked as Stable.
+
+At the end of this phase we want to achieve core APIs stability. This will allow developers to implement internal components and extend the collector will be marked as stable. The complete list of the packages/modules will be finalized during the first action item of this phase, but the tentative list is:
+
+* `consumer`
+  * Official internal data model `pdata`.
+  * Interfaces and utils to build a Consumer for (trace, metrics, logs).
+* `config`
+  * Core `config` including service definition, internal components definition will be stabilized.
+  * To be determined which config helpers will be marked as stable (e.g. configgrpc, etc.).
+* `component`
+  * Interfaces and utils to build a Collector internal component (receiver, processor, exporter, extension).
+* `obsreport`
+  * Focus on the public API of this package. It is out of scope to ensure stability for the metrics emitted (focus in phase 2).
+* `service`
+  * Public API to construct a OpenTelemetry Collector Service.
+
+**Action Items:**
+
+* Create a new milestone for this phase, create issues for all the other action items and add them to the milestone.
+* Agreement on all packages/modules that will be marked as stable during this phase.
+* Write a version doc as per [version and stability document](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md).
+  * Previously it was discussed that for the Collector it is fine to release stable golang modules that contain APIs marked as experimental.
```

Comment:
as long as it's clear what's supported and what's experimental when consuming the API. Having an external doc listing what's supported and what's not doesn't count. Like, the `spi` package could contain an `experimental` sub-package. This would be similar to Kubernetes using the apiVersion to mark what's stable (v1alpha1, v1beta1, v1, ...)

Reply:
I think it will be easier this way and do the work for all of them, otherwise we have more work to do when signals are getting stable. Pdata is autogenerated so not too much work to do now